src/realm/object-store/sync/sync_session.cpp

```diff
@@ -699,18 +715,24 @@ void SyncSession::handle_error(SyncError error)
         auto simplified_error = sync::websocket::get_simplified_websocket_error(
             static_cast<sync::websocket::WebSocketError>(error_code.value()));
         std::error_code new_error_code(simplified_error, sync::websocket::websocket_error_category());
-        error = SyncError(new_error_code, error.reason(), error.is_fatal);
+        error = sync::SessionErrorInfo(new_error_code, error.message, error.try_again);
     }
     else {
         // Unrecognized error code.
-        error.is_unrecognized_by_client = true;
+        unrecognized_by_client = true;
     }
 
     util::CheckedUniqueLock lock(m_state_mutex);
+    SyncError sync_error{error.error_code, std::string(error.message), error.is_fatal(), error.log_url,
+                         std::move(error.compensating_writes)};
+    // `action` is used over `shouldClientReset` and `isRecoveryModeDisabled`.
+    sync_error.server_requests_action = error.server_requests_action;
+    sync_error.is_unrecognized_by_client = unrecognized_by_client;
+
     if (delete_file)
-        update_error_and_mark_file_for_deletion(error, *delete_file);
+        update_error_and_mark_file_for_deletion(sync_error, *delete_file);
 
-    if (m_state == State::Dying && error.is_fatal) {
+    if (m_state == State::Dying && !error.try_again) {
```

Comment:
nit: you could either use `error.is_fatal()` or `sync_error.is_fatal`. Negating try_again does not clearly convey the intent imo.